addons/portal/models/mail_thread.py

```diff
@@ -87,3 +109,23 @@ def _portal_get_parent_hash_token(self, pid):
         :return: False or logical parent's _sign_token() result
         """
         return False
+
+    @api.model
+    def _get_thread_with_access(self, thread_model, thread_id, token, _hash, pid):
+        if _check_thread_portal_access(self.env, thread_model, thread_id, token, _hash, pid):
+            return (
+                super(MailThread, self.sudo())
+                ._get_thread_with_access(thread_model, thread_id, token, _hash, pid)
+                .with_context(mail_create_nosubscribe=True)
+            )
+        return super()._get_thread_with_access(thread_model, thread_id, token, _hash, pid)
+
+    def _prepare_post_data(self, post_data, special_mentions, **kwargs):
```

Comment:
This should be done with a classic override of ``message_post``, as this is the model method that handle posting a message. Everything that is model dependent for a model should be done in the right API method, otherwise we end up with code being managed partly in controller, partly in tools, partly in API methods.

Reply:
This is a route/portal specific override to enforce some values, do we really want to run this check on every other call to the method?

Actually I have the answer, if we have lower-level call to message post in some flows, we don't want them to be impacted by this code, even if the `portal_partner` is in the context. Only the top-most call done directly in the controller of message post should have this code check.

Reply:
Then probably to move in a controller. This PR mixes a lot of controller code and model code...